contractcourt/breacharbiter.go

```diff
@@ -643,6 +652,29 @@ func updateBreachInfo(breachInfo *retributionInfo, spends []spend) (
 			breachedOutput.outpoint, breachInfo.chanPoint)
 
 		doneOutputs[s.index] = struct{}{}
+
+		// Send report of the resolved output if there is a subscription.
+		if n, ok := b.subscriptions[breachInfo.chanPoint]; ok {
```

Comment:
Instead of this, I think it would be more elegant to append to a slice stored in `b.subscriptions` and send them all at the end in `notifyBreachComplete` and avoid the wait groups, goroutines. Then the breachresolver doesn't have to wait for a closed channel and can instead wait to receive the reports in a batch rather than incrementally. If for some reason the reports are empty, that should be handled also.

Reply:
It is also possible here that the counter-party manages to spend their breached outputs, whether it's their to_local output or their second-level htlc outputs. It would be rare as lnd would need to be down, borked, etc, but then the accounting would be incorrect which would be the least of your worries. I think adding logic here would be a nice future improvement, maybe add a comment?